src/plugins/vis_types/timeseries/public/convert_to_lens/lib/xy/date_histogram.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
```

Comment:
Splitting with terms and splitting with filters is not only a feature of the xy charts (for example a metric can also be split by filters or terms). So with the logic you are keeping on this PR they should not be part of the xy folder

Reply:
XY folder is about the logic of converting TSVB (standardized state) to Lens visualizations, which are not standardized in such a way. Like, we have two layers, first - extracting necessary data from TSVB standard state, and second - transforming it into some visualization with its own state. So, all the logic of transforming some data to Lens vis state is encapsulated in this folder. As XY is the only vis, which is relying on date_histogram, it is located in this folder. Possibly, I'm missing something. Could you, please, explain, what is wrong from your point of view? I'm opened to suggestions 😃

Reply:
If later some other vis will be added and it will use some features from xy, will move that functionality to the separate folder, out of XY.

Reply:
Thw date_histogram file has to do with splitting (grouping in TSVB). It doesnt have to do with the date_histogram agg that is used in timeseries chart. Moreover filters also applies to splitting. We know that this functionality (the grouping) applies to all other viz types. Why is it under xy? This is what I dont get!

Reply:
Maybe there's something I'm missing but I agree with @stratoula that this file as well as `filters.ts` and `metrics.ts` will be required for gauges/metrics and top n visualizations too so we should keep them in a central place.

Reply:
@stratoula I moved files, could you please review?